### PR TITLE
Generate email report for data registrar import jobs

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

Processors and thesis admins need to see summary data about the
registrar import job, especially if there are any issues with it.
The job currently logs much of this information, but an email report
would be more useful for stakeholders to review.

#### Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/ETD-354
* https://mitlibraries.atlassian.net/browse/ETD-348
* https://mitlibraries.atlassian.net/browse/ETD-371

#### How this addresses that need:

This adds some data collection to the import job, as well as a mailer
that reports the following data:

* Number of theses in the job
* Number of theses processed
* Errors found (total number and list)
* New DLCs added (total number and list)
* New degrees added (total number and list)
* Number of new users
* Number of new theses
* Number of updated theses

#### Side effects of this change:

* The email is sent in the job rather than the controller so as to
access the results, but we may need to move that up to the controller
if it doesn't work.
* For new DLCs and degrees, we are dumping the results of the `inspect`
method, which may be difficult to read for end users. We should take a
look at this in the PR build before merging.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [ ] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: app/views/report_mailer/registrar_import_email.html.erb [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
   <ul>
     <% @results[:new_degrees].each do |degree| %>
-      <li><%= degree %></li>
+      <li><%= link_to degree.name_dw, "/admin/degrees/#{degree.id}" %></li>
```
Comment (JPrevost) [inline app/views/report_mailer/registrar_import_email.html.erb:34]: I think these links will be problematic in an actual sent email (haven't confirmed yet).

We may need to look into this:
https://edgeguides.rubyonrails.org/action_mailer_basics.html#generating-urls-in-action-mailer-views

File: test/models/thesis_test.rb [hunk 2]
```diff
@@ -752,4 +752,17 @@ class ThesisTest < ActiveSupport::TestCase
     change = t.versions.last
     assert_equal change.changeset['title'], %w[MyString updated]
   end
+
+  test 'evaluates whether a thesis is newly created' do
+    preexisting_thesis = theses(:one)
+    assert_equal false, preexisting_thesis.new_thesis?
```
Comment (JPrevost) [inline test/models/thesis_test.rb:758]: You may want to consider using syntax like  `refute(preexisting_thesis.new_thesis?)` for true / false checks in mini_test.

https://minitest.rubystyle.guide/#refute-false

and 

https://minitest.rubystyle.guide/#assert-truthy

File: app/views/report_mailer/registrar_import_email.html.erb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<p>Hello,</p>
+
+<p>Below is a report of the registrar data import job submitted on 
+  <%= @registrar.created_at.in_time_zone("America/New_York").strftime('%b %-d, %Y at %l:%M %p %Z') %>.</p>
+
+<p>Summary of processed theses:</p>
+
+<ul>
+  <li>Total theses in job: <%= @results[:read] %>
```
Comment (JPrevost) [inline app/views/report_mailer/registrar_import_email.html.erb:9]: couple of missing `</li>` in this block

**Review by JPrevost — CHANGES_REQUESTED** [review]
A couple of missing `</li>` in the template and some suggested (not required) ideas on how to assert/refute true/false in tests.

File: app/mailers/report_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class ReportMailer < ApplicationMailer
+  def registrar_import_email(registrar, results)
+    return unless ENV.fetch('DISABLE_ALL_EMAIL', 'true') == 'false' # allows PR builds to disable emails
+
+    @registrar = registrar
+    @results = results
+    mail(from: "MIT Libraries <#{ENV['THESIS_ADMIN_EMAIL']}>",
+         to: ENV['THESIS_ADMIN_EMAIL'],
```
[the remark this answers is not in the document]
Comment (jazairi) [reply app/mailers/report_mailer.rb:8]: Good call. This reminds me that production.rb now includes a `MAILER_URL` variable that I don't think we need, so I'll remove that as well.

[the remark this answers is not in the document]
Comment (JPrevost) [reply app/mailers/report_mailer.rb:8]: Hmmm. I think we do need that right so we can set the URL to the preferred URL and not the herokuapp version for staging/prod?

[the remark this answers is not in the document]
Comment (JPrevost) [reply app/mailers/report_mailer.rb:8]: Oh, I see. You used preferred domain. never mind. Good idea.

[the remark this answers is not in the document]
Comment (jazairi) [reply app/mailers/report_mailer.rb:8]: `PREFERRED_DOMAIN` exists in staging and prod for that purpose. It's currently the first choice for the default_url config in production.rb, so the URLs should appear as expected.

**Review by JPrevost — APPROVED** [review]

**Pull request merged**